csrc/host_ir/jit.cpp

```diff
@@ -760,14 +862,28 @@ class HostIrCompileDispatcher : public OptInDispatch {
 
     // Convert input TensorViews to void pointers and get tensor pointers
     llvm::SmallVector<llvm::Value*, 1> input_tensors;
-    for (auto* tv : launch_kernel->inputs()) {
-      input_tensors.push_back(getOrCreateValue(tv, val_to_value_, builder_));
+    for (auto* in : launch_kernel->inputs()) {
+      if(auto* tv = in->as<TensorView>()) {
```

Comment:
Unlike `dynamic_cast`, `as` fails if `in` is not a TensorView: https://github.com/NVIDIA/Fuser/blob/a4692412787d5567e9ecda1a41a2ae6ef8e61549/csrc/utils.h#L117